lib/ansible/plugins/inventory/netbox.py

```diff
@@ -277,27 +287,36 @@ def extract_manufacturer(self, host):
     def extract_primary_ip(self, host):
         try:
             address = host["primary_ip"]["address"]
-            return str(ip_interface(address).ip)
+            return to_text(ip_interface(address).ip)
         except Exception:
             return
 
     def extract_primary_ip4(self, host):
         try:
             address = host["primary_ip4"]["address"]
-            return str(ip_interface(address).ip)
+            return to_text(ip_interface(address).ip)
         except Exception:
             return
 
     def extract_primary_ip6(self, host):
         try:
             address = host["primary_ip6"]["address"]
-            return str(ip_interface(address).ip)
+            return to_text(ip_interface(address).ip)
         except Exception:
             return
 
     def extract_tags(self, host):
         return host["tags"]
 
+    def extract_interfaces(self, host):
+        try:
+            if self.interfaces:
+                url = urljoin(self.api_endpoint, "/api/dcim/interfaces/?limit=0&device_id=%s" % (to_text(host["id"])))
```

Comment:
You do not differentiate between dcim and VMs, so this would raise a similar bug as #53393.

Reply:
This was a conscious decision, and probably short sighted.  It would be trivial to include it.